### DISCUSSION TITLE
testing/next - fabulous work!

### DISCUSSION THREAD

**Discussion by richb-hanover** [sqm-autorate/sqm-autorate#137] [Operational Development]
@chas-iot (and others?) Thanks for pulling together version 0.5.0 into the `testing/next` branch.

What's the next step for rolling it out and publicizing it? (I note that the `main` branch's README points to `testing/lua-threads`...) THANKS!

**Comment by chas-iot** [comment]
I'll leave it for review for another 12-18 hours in case @dlakelan, @jeverley, @Lochnair, or you @richb-hanover have any comments.
Then I'll merge into `testing/lua-threads` and announce on OpenWRT forums

**Comment by richb-hanover** [comment]
Perfect! (As a matter of policy, I might let it percolate for 24 hours - "letting it run overnight" has always worked well for me as a good first test for stability), but I agree with this plan.

I see the release notes show a number of changes to the implementation. Are all the relevant ones reflected in the README? 

I'll install it now (modifying the setup script invocation to use `testing/next`), watch for surprises, and will report tomorrow morning (it's almost 8am here.)

Thanks again.

**Reply by richb-hanover** [reply]
> I'll install it now (modifying the setup script invocation to use testing/next), 

Dang. This seems to have pulled in version 0.4.5 of the `sqm-autorate.lua` script...

Thoughts on how to test everything in the `testing/next` branch from start to finish without making it the "public branch"? Thanks.

```
root@D-Link-DIR-878:~# sh -c "$(wget -q -O- https://raw.githubusercontent.com/sq
m-autorate/sqm-autorate/testing/next/sqm-autorate-setup.sh)"
>>> Refreshing package cache. This may take several minutes...
>>> Installing required packages via opkg...
>>> Installing required packages via luarocks...
Installing https://luarocks.org/vstruct-2.1.1-1.src.rock...
Using https://luarocks.org/vstruct-2.1.1-1.src.rock... switching to 'build' mode
Updating manifest for /usr/lib/luarocks/rocks
vstruct 2.1.1-1 is now built and installed in /usr

>>> Pulling down sqm-autorate operational files...
>>> Putting sqm-autorate lib files into place...
>>> Making sqm-autorate.lua and getstats.sh executable...
>>> Putting config file into place...
>>> Putting service file into place...
> All done! You can enable and start the service by executing 'service sqm-autorate enable && service sqm-autorate start'.
```

**Comment by chas-iot** [comment]
I should have mentioned that it has already been reviewed by @Fail-Safe and has his OK, after some changes to the release notes. It's successfully run on his router and has been running on mine for more than 8 hours.

**Comment by richb-hanover** [comment]
> I'll install it now (modifying the setup script invocation to use testing/next), 

Dang. This seems to have pulled in version 0.4.5 of the `sqm-autorate.lua` script...

Thoughts on how to test everything in the `testing/next` branch from start to finish without making it the "public branch"? Thanks.

```
root@D-Link-DIR-878:~# sh -c "$(wget -q -O- https://raw.githubusercontent.com/sq
m-autorate/sqm-autorate/testing/next/sqm-autorate-setup.sh)"
>>> Refreshing package cache. This may take several minutes...
>>> Installing required packages via opkg...
>>> Installing required packages via luarocks...
Installing https://luarocks.org/vstruct-2.1.1-1.src.rock...
Using https://luarocks.org/vstruct-2.1.1-1.src.rock... switching to 'build' mode
Updating manifest for /usr/lib/luarocks/rocks
vstruct 2.1.1-1 is now built and installed in /usr

>>> Pulling down sqm-autorate operational files...
>>> Putting sqm-autorate lib files into place...
>>> Making sqm-autorate.lua and getstats.sh executable...
>>> Putting config file into place...
>>> Putting service file into place...
> All done! You can enable and start the service by executing 'service sqm-autorate enable && service sqm-autorate start'.
```

**Comment by richb-hanover** [comment]


**Reply by chas-iot** [reply]
try
```bash
wget -q -O- https://raw.githubusercontent.com/sqm-autorate/sqm-autorate/testing/next/sqm-autorate-setup.sh
./sqm-autorate-setup.sh testing/next
```
or `curl` instead of `wget`

**Comment by richb-hanover** [comment]
Would it work to force the branch as a parameter to the script? Like this?

```
sh -c "$(wget -q -O- https://raw.githubusercontent.com/sqm-autorate/sqm-autorate/testing/next/sqm-autorate-setup.sh testing/next ) "
```

**Reply by chas-iot** [reply]
If you work out how to do that, then I'll add it to the documentation ;D

**Reply by richb-hanover** [reply]
No not like that. I think it would be:

```
sh -c "$(wget -q -O- https://raw.githubusercontent.com/sqm-autorate/sqm-autorate/testing/next/sqm-autorate-setup.sh) "  testing/next 
```

**Reply by chas-iot** [reply]
did that work?

**Comment by richb-hanover** [comment]
Yes - using wget to retrieve the `setup.sh` script, then running it with a parameter of `testing/next` worked.

I see a regression though in the speed settings, especially for download. See Issue #138
